### DISCUSSION TITLE
Feature request: Markdown - Custom title for Alerts

### DISCUSSION THREAD

**Discussion by jbee** [community/community#103219] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

What I would like to do is a slight extension to the alerts

> [!Important]
> :point_up: I want to change the label used here
 
My suggestion for the markup would be something like

    > [!Important]: My Title
    > Hello
    
which should render as

> **My Title** 
> Hello

But in the styling of the `!Important` alert.

**Comment by ghost** [comment]
Yeah that would awesome and helpful

**Comment by ruby-dev** [comment]
This would be a great addition and help with formatting documents much easier.

**Comment by luislvasquez** [comment]
I would love this to be implemented. In my case, PR descriptions would look way better with this

**Comment by FeralFlora** [comment]
This is supported in the Obsidian flavour of markdown, although without needing the `:`. So this would bring the two syntaxes closer together:
https://help.obsidian.md/Editing+and+formatting/Callouts#Change+the+title

**Reply by mkurz** [reply]
+1000 please please use the Obsidian syntax if you ever implement this feature.

**Reply by jbee** [reply]
I really don't care which syntax it is as long as we get the feature :)

**Reply by mkurz** [reply]
I do care which syntax it is, it would be great if they stay compatible.

**Reply by jbee** [reply]
I there any existing integration or interaction between GH and obsidian or are you just wanting this out of convenience so you don't have to @mkurz remember two sets of markup as you are using both?

**Reply by mkurz** [reply]
I think it just makes to keep it compatible. Compatibility is always good. E.g. I would prefer to write longe comments I want to add to a GitHub discussion or gists or GitHub wikis in markdown in obisidanj because I just like their wysiwyg editor and then just paste that into GitHub.

**Comment by willlockwood** [comment]
Commenting in support of this request—I have often found myself wanting to use the Github markdown alert style in my docs, but _every single time_ I have not used them, because the baked-in titles for the alerts don't feel like they match the context of the document I'm writing.

For example—these are all alerts that I'd be excited to use in READMEs, design docs, and even in PR descriptions, if this kind of customization were available:

<img width="538" alt="Screenshot 2024-08-16 at 7 35 52 PM" src="https://github.com/user-attachments/assets/75d6861a-0e4b-4973-9cda-b9cb7103c966">

**Reply by jbee** [reply]
I had similar things in mind :)

**Reply by null9beta** [reply]
This simple change would be awesome and make the callouts so much more usable. The pre-defined wordings are fine as a fallback, though the suggested addition exactely for the purpose you described @willlockwood would be a blast

**Comment by p4gs** [comment]
Would also love this!

**Comment by huaminghuangtw** [comment]
Second this!

**Comment by 5HT2** [comment]
Any news? It would be great to see this added.

**Comment by tlourey** [comment]
I agree, this would be great to have added.

**Comment by algoldst** [comment]
Would also like this feature. Right now the only way to get titles on MkDocs callouts / admonitions / alerts is to use their 

> NOTE: **TITLE**
> Note text

While they support GitHub-style alerts, they don't support adding a title because, so far, GitHub doesn't either.

This is especially annoying because Obsidian **does** support titles, and does this very well.

> [!NOTE] Title
> Note text

So this should be easily able to be implemented. And for the love of god, please use Obsidian's format to implement it — there's no sense in having multiple different "standards" for authors to remember & maintain.

**Comment by falkgeist** [comment]
Just as an addition: Right now, the titles don't translate e.g. on a repositories wiki page. And I think I saw another discussion saying it doesn't work in Github Pages.

Which this feature would at least create a workaround for.

So, I definitely say "YES" to this feature.

**Reply by jbee** [reply]
Thanks, we might only have to wait 2 more years to get a 5 min coding exercise implemented that I bet any coding AI even does for you in no time flat. What a time to be alive... waiting...

**Comment by calacayir** [comment]
If this feature will be implemented, I recommend using this format: 

```
> [!note]
> **Note Title**
> Note Text
```

i.e. use bold text in the first line as the title of the alert.
This is because this block will still be rendered sensibly with markdown renderers that do not support this feature yet.
For example it looks like this when not supported:

> [!note]
> **Note Title**
> Note Text

I am currently using this format in my docs generation project [docfx-plus](https://github.com/dotmake-build/docfx-plus).

**Comment by tylerharris82836-netizen** [comment]
That extension makes sense and would improve clarity without breaking existing alert behavior; a custom title while keeping the !Important styling feels flexible and readable. The approach reminds me of how structured tagging is handled in systems managing things like [final expense insurance leads](https://lifeinsuranceleadgenagency.co.uk/services/final-expense/), where labels add context without changing the core format.
